internal/pkg/comm/server.go

```diff
@@ -73,11 +73,15 @@ func NewGRPCServerFromListener(listener net.Listener, serverConfig ServerConfig)
 	if secureConfig.UseTLS {
 		//both key and cert are required
 		if secureConfig.Key != nil && secureConfig.Certificate != nil {
+			grpcServer.tls = &TLSConfig{
```

Comment:
you have a constructor method, yet you create the instance without it.

Let's be consistent.